MACE_Frontend/src/components/map/index.tsx

```diff
@@ -65,29 +73,78 @@ export default class MapView extends React.Component<Props, State> {
     if(this.state.contextMenuVisible != nextState.contextMenuVisible) {
         return true;
     }
+    if(this.state.popupVisible != nextState.popupVisible) {
+      return true;
+    }
     return false;
   }
 
+  togglePopup = () => {
+    this.setState({selectedAircraft: "ALL"});
+    this.setState({ popupVisible: !this.state.popupVisible });
+  }
+
+  onAircraftSelected = (e) => {
+    this.setState({selectedAircraft: e.target.value });
+  }
+  createDropdownList = () => {
+    let options = [];
+    let hide = false;
+    options.push(<option key={0} value="ALL">All Aircraft</option>);
+    this.props.context.aircrafts.forEach((a) => {
+      if (["STABILIZE",""].indexOf(a.mode) != -1){
+        hide = true;
+      } else {
+        hide = false;
+      }
+      options.push(<option key={options.length} value={a.agentID} disabled={hide}>{"Agent " + a.agentID}</option>);
+    });
+    return options;
+  }
+  handlevehicleHome = () => {
+    this.setState({ popupVisible: false });
+    let command: string = "SET_VEHICLE_HOME";
+    let payload = {
+      lat: this.state.originPosition.lat,
+      lng: this.state.originPosition.lng,
+      alt: 0.0
+    };
+    let payloadArray = [];
+    let enabledAircraft = [];
+    this.props.context.aircrafts.forEach((a) => {
+      if (["STABILIZE",""].indexOf(a.mode) == -1){
```

Comment:
Similar question to a previous comment: why check for the `STABILIZE` mode? I think if "ALL" is selected, then the home should be sent for every vehicle, not just those in `STABILIZE` mode. If specific modes are required for setting home positions, I think we would have to go one step further and actually require a "GROUNDED" state, for which we would need the MACE state machine status to properly handle.